app/services/creators/employments_creator.rb

```diff
@@ -1,9 +1,10 @@
 module Creators
   class EmploymentsCreator < BaseCreator
-    def initialize(assessment_id:, employments_params:)
+    def initialize(assessment_id:, employments_params:, employment_collection: nil)
       super()
       @assessment_id = assessment_id
       @employments_params = employments_params
+      @explicit_employment_collection = employment_collection
```

Comment:
IMHO it would be nicer if the caller always passed this parameter, rather than this class taking an optional parameter and needing a method to resolve it. Was there no simple way of taking an assessment object rather than an assessment id?

Reply:
@starswan the issue here is that in the normal run of things the 'dumb' controller passes the params straight onto these creators. The creator then handles behaving appropriately if, for example, an invalid assessment_id param is passed in.

If we insist that the creators receive an explicit summary, then before calling the creator, something (please please not the controller itself), has to instantiate an assessment object and handle something going wrong if that fails. Which means adding an extra layer of objects between the controllers and creators, which becomes quite a disruptive change. 

My feeling was to try to keep a lighter touch than that, to make it less impossible to unfork, down the line. WDYT?

Reply:
@patrick-laa I was expecting that the controller would always instantiate an 'assessment' object (in the before action) but it looks like a different pattern has been used here +1